src/Domain/CoordinateMaps/Frustum.cpp

```diff
@@ -219,6 +233,16 @@ std::optional<std::array<double, 3>> Frustum::inverse(
   if (zeta_distribution_ == Distribution::Projective) {
     logical_coords[2] = (-w_minus_ + w_plus_ * logical_coords[2]) /
                         (w_plus_ - w_minus_ * logical_coords[2]);
+  } else if (zeta_distribution_ == Distribution::Logarithmic) {
+    const double sigma_r = 0.5 * (inner_radius_ + radius_);
```

Comment:
I suggest you give `domain::CoordinateMaps::Interval` a try. If you can use that to redistribute grid points on the logical axis before applying the frustum map then it would make the code here a lot simpler.